### ISSUE TITLE
Port twisted.internet.threads to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6067]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6067|
|Type|enhancement|
|Created|2012-10-05 19:19:44Z|
|Branch|https://github.com/twisted/twisted/tree/threads-py3-6067|

`twisted.internet.threads` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6067 6067
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_threads_py3_6067 branches/threads-py3-6067
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349464784000000 1349464784000000
changetime__1349798962000000 1349798962000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35989]) Branching to 'threads-py3-6067'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/threads-py3-6067 was started.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Great, thanks, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [36024]) Merge threads-py3-6067.

Author: itamar
Review: exarkun
Fixes: [#6067](https://github.com/twisted/twisted/issues/6067)

twisted.internet.threads and IReactorThreads now work on Python 3.
